tests/automation-framework/rancherclient/rancherclient.go

```diff
@@ -0,0 +1,79 @@
+package rancherclient
+
+import (
+	projectClient "github.com/rancher/rancher/tests/automation-framework/client/generated/project/v3"
+	"github.com/rancher/rancher/tests/automation-framework/clientbase"
+	"github.com/rancher/rancher/tests/automation-framework/config"
+	managementClient "github.com/rancher/rancher/tests/automation-framework/management"
+	"github.com/rancher/rancher/tests/automation-framework/testsession"
+	"k8s.io/client-go/rest"
+)
+
+type RancherClient struct {
+	Management    *managementClient.Client
+	Project       *projectClient.Client
+	RestConfig    *rest.Config
+	RancherConfig *config.RancherServerConfig
+}
+
+// NewClient is returns a larger client wrapping individual api clients.
+func NewClient(restConfig *rest.Config, testSession *testsession.TestSession) (*RancherClient, error) {
+	c := &RancherClient{
+		RestConfig: restConfig,
+	}
+
+	err := c.newManagementClient(testSession)
+	if err != nil {
+		return nil, err
+	}
+
+	err = c.newProjectClient()
+	if err != nil {
+		return nil, err
+	}
+	c.Project.APIBaseClient.Ops.TestSession = testSession
+
+	return c, nil
+}
+
+// NewRestConfig is the config used the various clients
+func NewRestConfig(bearerToken string, rancherConfig *config.RancherServerConfig) *rest.Config {
```

Comment:
can we move bearerToken into the config object?

Reply:
I'm doing is this way, for cases with multiple bearer tokens ie) test cases with an admin user, and a standard user. The config object has attributes for AdminToken, and UserToken. Which is something we have test cases planned for automation.

Reply:
I see, so your thinking is a developer who has a test with multiple users would create a new client for each user?